geofence/geofence.go

```diff
@@ -0,0 +1,66 @@
+package echogeofence
+
+import (
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/circa10a/go-geofence"
+	"github.com/labstack/echo/v4"
+)
+
+// EchoGeofence holds a geofence config
+type EchoGeofence geofence.Config
+
+// config returns a geofence config from go-geofence
+func (e *EchoGeofence) config() *geofence.Config {
+	return &geofence.Config{
+		IPAddress: e.IPAddress,
+		Token:     e.Token,
+		Radius:    e.Radius,
+		CacheTTL:  e.CacheTTL,
+	}
+}
+
+// Middleware looks up the coordinates of a client to see if it's nearby
+func Middleware(echoGeofence *EchoGeofence) echo.MiddlewareFunc {
+	return func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(c echo.Context) error {
+			// Get client ip, strip port
+			remoteAddr := c.Request().RemoteAddr
+			ipAddress, _, err := net.SplitHostPort(remoteAddr)
+			if err != nil {
+				return err
+			}
+
+			// Ensure not a private ip
+			ip := net.ParseIP(ipAddress)
+			if ip.IsPrivate() || strings.HasPrefix(ipAddress, "::1") {
+				return next(c)
+			}
+
+			// Create new geofence
+			geofence, err := geofence.New(echoGeofence.config())
```

Comment:
This could probably moved up and be created only once (I assume it is coroutine safe object) (if not using proposed interface)
```
	return func(next echo.HandlerFunc) echo.HandlerFunc {
                geofence, err := geofence.New(echoGeofence.config())
		return func(c echo.Context) error {
```

Reply:
I think this is no obsolete after implemented the interface changes you requested below.